components/org.wso2.carbon.stream.processor.statistics/src/main/java/org/wso2/carbon/stream/processor/statistics/impl/StatisticsApiServiceImpl.java

```diff
@@ -49,8 +53,7 @@ public Response statisticsGet() {
             return Response.status(Response.Status.OK).entity(osMetricsJSON).build();
         } catch (Exception e) {
             String message = e.getMessage();
-            if(("Wso2 Carbon metrics is not enabled.".equalsIgnoreCase(message)) || ("JMX reporter is not running. Please enable the JMX reporter at " +
-                    "carbon metrics.").equalsIgnoreCase(message)) {
+            if(("Wso2 Carbon metrics is not enabled.".equalsIgnoreCase(message)) || ("JMX reporter has been disabled at WSO2 carbon metrics.").equalsIgnoreCase(message)) {
```

Comment:
Better to have different exception types than checking message

Reply:
Fixed in 9c0442d118e0b259286b6199b2b5afcb8c905b6c